core/blockchain_test.go

```diff
@@ -1998,6 +1171,119 @@ func testInsertKnownChainData(t *testing.T, typ string) {
 	asserter(t, blocks2[len(blocks2)-1])
 }
 
+func TestInsertDifferentHeadersAfterReset(t *testing.T) {
+	testInsertDifferentChainDataAfterReset(t, "headers", 32, 64)
+}
+func TestInsertDifferentReceiptChainAfterReset(t *testing.T) {
+	testInsertDifferentChainDataAfterReset(t, "receipts", 32, 64)
+}
+func TestInsertDifferentBlocksAfterReset(t *testing.T) {
+	testInsertDifferentChainDataAfterReset(t, "blocks", 32, 64)
+}
+
+func testInsertDifferentChainDataAfterReset(t *testing.T, typ string, setHeadBlock int, chainsBlocksNumber int) {
+	// log.Root().SetHandler(log.LvlFilterHandler(log.LvlTrace, log.StreamHandler(os.Stderr, log.TerminalFormat(true))))
+
+	engine := mockEngine.NewFaker()
+
+	db := rawdb.NewMemoryDatabase()
+	genesis := new(Genesis).MustCommit(db)
+
+	originalBlocks, originalReceipts := GenerateChain(params.TestChainConfig, genesis, engine, db, chainsBlocksNumber, func(i int, b *BlockGen) {
+		b.SetCoinbase(common.Address{1})
+	})
+	// Total difficulty is higher to create a different chain.
```

Comment:
How is the total difficulty higher here, and why is it important?

Reply:
Haha, yes, sorry
Changing the comment to:
`// Block offsetTime different to create a different chain.`
Basically if I didn't change something in the generation of the block, it generates the same exactly chain